### PR TITLE
Feat/4703  search tags

### PR DESCRIPTION
**Pull request by grignards**
See #4703.

## Checkpoints

**For developers**

- [x] If relevant, manual tests have been done to ensure the stability of the whole application and that the involved feature works
- [x] The original issue is up to date w.r.t the latest discussions and contains a short summary of the implemented solution
- [x] Automated tests covering the feature or the fix, have been written, deemed irrelevant (give the reason), or an issue has been created to implement the test (give the link)
- [x] Make sure that:
  - if there are modifications in the Tracim configuration files (eg. `development.ini`), they are documented in `backend/doc/setting.md`
  - any migration process required for existing instances is documented
  - relevant people for these changes are notified
- [x] Original authors of the features included in a multi-feature branch (maintenance fixes -> develop, security fixes -> develop, …) should be part of the reviewers, especially if you encountered merge conflicts.

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, related issues are mentioned in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present

**For testers**

- [x] Manual, quality tests have been done

### CODE REVIEW

**Review by gbonaspetti — APPROVED** [review]

**Comment by gbonaspetti** [comment]
Frontend: approved

File: backend/tracim_backend/lib/core/tag.py [hunk 1]
```diff
@@ -146,3 +146,12 @@ def delete(self, tag: Tag, do_save: bool) -> None:
         self._session.delete(tag)
         if do_save:
             self._session.flush()
+
+    def get_contents(self, tag: Tag) -> typing.List[Content]:
+        """Return the list of contents which includes the given tag."""
```
Comment (raphj) [inline backend/tracim_backend/lib/core/tag.py:151]:
```suggestion
        """Return the list of contents which include the given tag."""
```

File: frontend/src/action-creator.async.js [hunk 2]
```diff
@@ -1167,6 +1167,7 @@ export const getAdvancedSearchResult = (
       if (searchFacets.content_types) queryParameterList.push(`content_types=${searchFacets.content_types}`)
       if (searchFacets.file_extensions) queryParameterList.push(`file_extensions=${searchFacets.file_extensions}`)
       if (searchFacets.author__public_names) queryParameterList.push(`author__public_names=${searchFacets.author__public_names}`)
+      if (searchFacets.tags) queryParameterList.push(`tags=${searchFacets.tags}`)
```
Comment (raphj) [inline frontend/src/action-creator.async.js:1170]: is searchFacets.tags an array? are the elements url encoded? Does this work with a tag containing a comma?

**Review by raphj — APPROVED** [review]

**Review by inkhey — APPROVED** [review]

File: frontend/src/action-creator.async.js (hunk 2 above)
Comment (grignards) [reply frontend/src/action-creator.async.js]: I just added proper array support.

File: frontend/src/container/AdvancedSearch.jsx [hunk 3]
```diff
@@ -251,28 +254,42 @@ export class AdvancedSearch extends React.Component {
...
 
-  updateAppliedFilter = (type, oldAppliedFilter, filterObject) => {
+  updateList = (value, oldValueList = []) => {
+    const newValueList = oldValueList.includes(value)
+      ? oldValueList.filter(f => f !== value)
+      : [...oldValueList, value]
+    return newValueList.length > 0 ? newValueList : null
+  }
+
+  updateScalar = (value) => value
```
Comment (gbonaspetti) [inline frontend/src/container/AdvancedSearch.jsx:269]: Maybe add a INFO comment about the need/use of this function

**Review by raphj — APPROVED** [review]

**Comment by lebouquetin** [comment]
@grignards if I understand correctly, the mechanism is to reindex content when a tag is added/deleted. It means for example that if I add 5 tags to a 50 pages document there will be 250 pages indexed in elastic search? Is that right?

**Comment by grignards** [comment]
> @grignards if I understand correctly, the mechanism is to reindex content when a tag is added/deleted. It means for example that if I add 5 tags to a 50 pages document there will be 250 pages indexed in elastic search? Is that right?

Yes, currently we do not have a proper mechanism to handle partial indexing of contents. @inkhey is more aware of the implications: AFAIK it is possible but more complicated so it wasn't implement for now.

**Comment by gbonaspetti** [comment]
Tested on branch

**Pull request merged**